### ISSUE TITLE
Merge Exchange functions

### ISSUE THREAD

**Issue by davidcortesortuno** [computationalmodelling/fidimag#39]
The Exchange class has two functions: `UniformExchange` and `Exchange`. We should merge them in a single Exchange function. The `Exchange` already checks if the exchange constant is a number or a function

**Comment by ww1g11** [comment]
Yes, we should merge them at some point but not now, because the Exchange class is not finished -- I am thinking to use a more friendly interface if the exchange tensor is varying spatially. Right now we just simply need a function with six components for each spin.
